homeassistant/components/system_bridge/__init__.py

```diff
@@ -328,3 +336,22 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 async def async_reload_entry(hass: HomeAssistant, entry: ConfigEntry) -> None:
     """Reload the config entry when it changed."""
     await hass.config_entries.async_reload(entry.entry_id)
+
+
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    _LOGGER.debug("Migrating from version %s", config_entry.version)
+
+    if config_entry.version == 1:
+        new = {
+            **config_entry.data,
+            CONF_TOKEN: config_entry.data.get(CONF_API_KEY),
+        }
+        if CONF_API_KEY in new:
+            new.pop(CONF_API_KEY)
+        config_entry.version = 2
+        hass.config_entries.async_update_entry(config_entry, data=new)
+
+    _LOGGER.debug("Migration to version %s successful", config_entry.version)
+
+    return True
```

Comment:
This doesn't look correct, maybe something like this to ensure migration back and forth between 1.1 and 1.2 works (I've not tried the suggested change, it's just a suggestion).

```suggestion
async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
    """Migrate old entry."""
    _LOGGER.debug("Migrating from version %s", config_entry.version)

    if config_entry.version == 1:
        if config_entry.minor_version == 1:
            # Migrate to CONF_TOKEN, which was added in 1.2
            new_data = dict(config_entry.data)
            new_data.setdefault(CONF_TOKEN, data[CONF_API_KEY]
        hass.config_entries.async_update_entry(config_entry, data=new, minor_version=2)

        _LOGGER.debug(
            "Migration to version %s.%s successful",
            config_entry.version,
            config_entry.minor_version,
        )

    # User is trying to downgrade from a future version
    return False
```